In [1]:
import os
os.chdir("D:\\UIT\\HK\\HK6\\IS403\\Đồ án\\Source code")
print("Thư mục làm việc hiện tại:", os.getcwd())

Thư mục làm việc hiện tại: D:\UIT\HK\HK6\IS403\Đồ án\Source code


# Import modules

In [2]:
import pandas as pd

In [3]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# Load data

In [4]:
vcb_kqkd = pd.read_csv(".\\dataset\\clean_dataset\\bank_index\\vcb\\vcb_kqkd.csv")
vcb_kqkd

Năm  Kỳ  Doanh thu (đồng)  Tăng trưởng doanh thu (%)  \
0   2024   4    23581262000000                  -0.060425   
1   2024   3    23191212000000                  -0.125269   
2   2024   2    22862707000000                  -0.206768   
3   2024   1    24019660000000                  -0.132547   
4   2023   4    25097791000000                  -0.017087   
5   2023   3    26512406000000                   0.174704   
6   2023   2    28822211000000                   0.394328   
7   2023   1    27689870000000                   0.431878   
8   2022   4    25534087000000                   0.423362   
9   2022   3    22569428000000                   0.279451   
10  2022   2    20671039000000                   0.158584   
11  2022   1    19338145000000                   0.124238   
12  2021   4    17939277000000                   0.045207   

    Lợi nhuận sau thuế của Cổ đông công ty mẹ (đồng)  \
0                                      8565378000000   
1                                      8566598000000   
2                                      8119414000000   
3                                      8579996000000   
4                                      9355026000000   
5                                      7268919000000   
6                                      7422823000000   
7                                      8986435000000   
8                                      9928277000000   
9                                      6064870000000   
10                                     5936954000000   
11                                     7962184000000   
12                                     6539788000000   

    Tăng trưởng lợi nhuận (%)   LN trước thuế  Lợi nhuận thuần  Cố tức đã nhận  
0                   -0.084409  10702677000000    8569921000000     66378000000  
1                    0.178524  10698944000000    8572494000000     49396000000  
2                    0.093844  10116275000000    8124709000000    159132000000  
3                   -0.045228  10718239000000    8585993000000     32273000000  
4                   -0.057739  11693497000000    9360033000000     50548000000  
5                    0.198528   9051164000000    7274556000000    146603000000  
6                    0.250275   9277682000000    7427958000000     29775000000  
7                    0.128639  11221344000000    8991901000000     39530000000  
8                    0.518134  12419095000000    9933519000000     32313000000  
9                    0.320135   7566222000000    6069444000000    127264000000  
10                   0.499191   7423087000000    5941931000000     24345000000  
11                   0.153509   9950256000000    7967426000000     24495000000  
12                   0.153907   8174561000000    6545349000000     31107000000

In [5]:
vcb_cstc = pd.read_csv(".\\dataset\\clean_dataset\\bank_index\\vcb\\vcb_cstc.csv")
vcb_cstc

CP   Năm  Kỳ  Biên lợi nhuận ròng (%)   ROE (%)  ROIC (%)   ROA (%)  \
0   VCB  2024   4                 0.618781  0.187316       0.0  0.017237   
1   VCB  2024   3                 0.630934  0.193913       0.0  0.018586   
2   VCB  2024   2                 0.583807  0.195129       0.0  0.018386   
3   VCB  2024   1                 0.609459  0.199735       0.0  0.018516   
4   VCB  2023   4                 0.730792  0.219739       0.0  0.018083   
5   VCB  2023   3                 0.577076  0.226347       0.0  0.018943   
6   VCB  2023   2                 0.529423  0.230197       0.0  0.018477   
7   VCB  2023   1                 0.632716  0.231573       0.0  0.017892   
8   VCB  2022   4                 0.670401  0.244309       0.0  0.018522   
9   VCB  2022   3                 0.443859  0.220298       0.0  0.017240   
10  VCB  2022   2                 0.463926  0.216549       0.0  0.017014   
11  VCB  2022   1                 0.664853  0.207517       0.0  0.016503   
12  VCB  2021   4                 0.613101  0.215723       0.0  0.015994   

    Tỷ suất cổ tức (%)  Vốn hóa (Tỷ đồng)  Số CP lưu hành (Triệu CP)  \
0                  0.0    501340505640000                 8355675094   
1                  0.0    512519668725400                 5589091262   
2                  0.0    510842941346800                 5589091262   
3                  0.0    489604394551200                 5589091262   
4                  0.0    528728033385200                 5589091262   
5                  0.0    495752394939400                 5589091262   
6                  0.0    480661848532000                 5589091262   
7                  0.0    492181723384000                 4732516571   
8                  0.0    418354464876400                 4732516571   
9                  0.0    406049921791800                 4732516571   
10                 0.0    314712351971500                 4732516571   
11                 0.0    339794689797800                 4732516571   
12                 0.0    382860590593900                 4732516571   

          P/E       P/B       P/S  P/Cash Flow    EPS (VND)    BVPS (VND)  
0   14.818799  2.556387  7.310462     8.479643  1025.097063  23470.623833  
1   14.803708  2.694830  7.669374   -10.421608  1532.735394  34028.114605  
2   15.329877  2.813058  7.767334   123.346014  1452.725250  32491.336872  
3   15.006220  2.821707  7.363946    -3.257646  1535.132564  31045.041468  
4   16.005957  3.205992  7.807162   -13.718398  1673.800903  29507.243355  
5   14.751702  3.115219  7.039211     6.687242  1300.554716  28473.118534  
6   14.834141  3.164272  6.735315    10.304262  1328.091214  27178.445454  
7   15.919692  3.404580  7.044594     1.981862  1898.870266  30547.090925  
8   13.992249  3.086173  6.144787     2.280837  2097.885311  28643.893786  
9   15.371801  3.165160  6.315958     8.228452  1281.531699  27107.637359  
10  12.616499  2.560887  5.173818     4.265981  1254.502528  25967.563168  
11  14.794486  2.904594  5.882990   -22.191178  1682.441864  24719.462942  
12  17.467214  3.511507  6.749586    -6.989035  1381.883804  23038.541622

In [6]:
vcb_cdtc = pd.read_csv(".\\dataset\\clean_dataset\\bank_index\\vcb\\vcb_cdtc.csv")
vcb_cdtc

Năm  Kỳ  TỔNG CỘNG TÀI SẢN (đồng)  TỔNG CỘNG NGUỒN VỐN (đồng)  \
0   2024   4          2085873522000000            2085873522000000   
1   2024   3          1932361908000000            1932361908000000   
2   2024   2          1905638840000000            1905638840000000   
3   2024   1          1773713584000000            1773713584000000   
4   2023   4          1839223339000000            1839223339000000   
5   2023   3          1731305572000000            1731305572000000   
6   2023   2          1704273133000000            1704273133000000   
7   2023   1          1846431290000000            1846431290000000   
8   2022   4          1814188209000000            1814188209000000   
9   2022   3          1649664327000000            1649664327000000   
10  2022   2          1601680964000000            1601680964000000   
11  2022   1          1462748158000000            1462748158000000   
12  2021   4          1414765480000000            1414765480000000   

    NỢ PHẢI TRẢ (đồng)  VỐN CHỦ SỞ HỮU (đồng)  LỢI ÍCH CỦA CỔ ĐÔNG THIỂU SỐ  
0     1889664354000000        196209168000000                   96261000000  
1     1742064907000000        190297001000000                  110763000000  
2     1723936055000000        181702785000000                  105738000000  
3     1600099458000000        173614126000000                  100556000000  
4     1670680015000000        168543324000000                   93990000000  
5     1572062054000000        159243518000000                  104660000000  
6     1552272005000000        152001128000000                   98316000000  
7     1701772932000000        144658358000000                   93744000000  
8     1676199767000000        137988442000000                   88379000000  
9     1521274777000000        128389550000000                  102207000000  
10    1478691267000000        122989697000000                   97774000000  
11    1345670470000000        117077688000000                   92420000000  
12    1303594825000000        111170655000000                   87113000000

# Perform descriptive statistics

In [7]:
vcb_kqkd.describe()

Năm         Kỳ  Doanh thu (đồng)  Tăng trưởng doanh thu (%)  \
count    13.000000  13.000000      1.300000e+01                  13.000000   
mean   2022.846154   2.615385      2.367916e+13                   0.114589   
std       0.987096   1.192928      3.150641e+12                   0.220180   
min    2021.000000   1.000000      1.793928e+13                  -0.206768   
25%    2022.000000   2.000000      2.256943e+13                  -0.060425   
50%    2023.000000   3.000000      2.358126e+13                   0.124238   
75%    2024.000000   4.000000      2.553409e+13                   0.279451   
max    2024.000000   4.000000      2.882221e+13                   0.431878   

       Lợi nhuận sau thuế của Cổ đông công ty mẹ (đồng)  \
count                                      1.300000e+01   
mean                                       7.945897e+12   
std                                        1.241063e+12   
min                                        5.936954e+12   
25%                                        7.268919e+12   
50%                                        8.119414e+12   
75%                                        8.579996e+12   
max                                        9.928277e+12   

       Tăng trưởng lợi nhuận (%)  LN trước thuế  Lợi nhuận thuần  \
count                  13.000000   1.300000e+01     1.300000e+01   
mean                    0.177485   9.924080e+12     7.951172e+12   
std                     0.189222   1.553728e+12     1.241163e+12   
min                    -0.084409   7.423087e+12     5.941931e+12   
25%                     0.093844   9.051164e+12     7.274556e+12   
50%                     0.153907   1.011628e+13     8.124709e+12   
75%                     0.250275   1.071824e+13     8.585993e+12   
max                     0.518134   1.241910e+13     9.933519e+12   

       Cố tức đã nhận  
count    1.300000e+01  
mean     6.255069e+10  
std      4.851476e+10  
min      2.434500e+10  
25%      3.110700e+10  
50%      3.953000e+10  
75%      6.637800e+10  
max      1.591320e+11

In [8]:
vcb_cstc.describe()

Năm         Kỳ  Biên lợi nhuận ròng (%)    ROE (%)  ROIC (%)  \
count    13.000000  13.000000                13.000000  13.000000      13.0   
mean   2022.846154   2.615385                 0.597625   0.214488       0.0   
std       0.987096   1.192928                 0.080364   0.016927       0.0   
min    2021.000000   1.000000                 0.443859   0.187316       0.0   
25%    2022.000000   2.000000                 0.577076   0.199735       0.0   
50%    2023.000000   3.000000                 0.613101   0.216549       0.0   
75%    2024.000000   4.000000                 0.632716   0.226347       0.0   
max    2024.000000   4.000000                 0.730792   0.244309       0.0   

         ROA (%)  Tỷ suất cổ tức (%)  Vốn hóa (Tỷ đồng)  \
count  13.000000                13.0       1.300000e+01   
mean    0.017799                 0.0       4.518003e+14   
std     0.000915                 0.0       7.110575e+13   
min     0.015994                 0.0       3.147124e+14   
25%     0.017237                 0.0       4.060499e+14   
50%     0.018083                 0.0       4.896044e+14   
75%     0.018516                 0.0       5.013405e+14   
max     0.018943                 0.0       5.287280e+14   

       Số CP lưu hành (Triệu CP)        P/E        P/B        P/S  \
count               1.300000e+01  13.000000  13.000000  13.000000   
mean                5.406563e+09  15.054796   3.000336   6.846503   
std                 9.841745e+08   1.119985   0.302944   0.792105   
min                 4.732517e+09  12.616499   2.556387   5.173818   
25%                 4.732517e+09  14.794486   2.813058   6.315958   
50%                 5.589091e+09  14.834141   3.086173   7.039211   
75%                 5.589091e+09  15.371801   3.165160   7.363946   
max                 8.355675e+09  17.467214   3.511507   7.807162   

       P/Cash Flow    EPS (VND)    BVPS (VND)  
count    13.000000    13.000000     13.000000  
mean      8.384341  1495.788660  28170.624148  
std      35.901398   288.845267   3370.179087  
min     -22.191178  1025.097063  23038.541622  
25%      -6.989035  1300.554716  25967.563168  
50%       2.280837  1452.725250  28473.118534  
75%       8.228452  1673.800903  30547.090925  
max     123.346014  2097.885311  34028.114605

In [9]:
vcb_cdtc.describe()

Năm         Kỳ  TỔNG CỘNG TÀI SẢN (đồng)  \
count    13.000000  13.000000              1.300000e+01   
mean   2022.846154   2.615385              1.750913e+15   
std       0.987096   1.192928              1.874442e+14   
min    2021.000000   1.000000              1.414765e+15   
25%    2022.000000   2.000000              1.649664e+15   
50%    2023.000000   3.000000              1.773714e+15   
75%    2024.000000   4.000000              1.846431e+15   
max    2024.000000   4.000000              2.085874e+15   

       TỔNG CỘNG NGUỒN VỐN (đồng)  NỢ PHẢI TRẢ (đồng)  VỐN CHỦ SỞ HỮU (đồng)  \
count                1.300000e+01        1.300000e+01           1.300000e+01   
mean                 1.750913e+15        1.598306e+15           1.526066e+14   
std                  1.874442e+14        1.625531e+14           2.832517e+13   
min                  1.414765e+15        1.303595e+15           1.111707e+14   
25%                  1.649664e+15        1.521275e+15           1.283896e+14   
50%                  1.773714e+15        1.600099e+15           1.520011e+14   
75%                  1.846431e+15        1.701773e+15           1.736141e+14   
max                  2.085874e+15        1.889664e+15           1.962092e+14   

       LỢI ÍCH CỦA CỔ ĐÔNG THIỂU SỐ  
count                  1.300000e+01  
mean                   9.784008e+10  
std                    6.895780e+09  
min                    8.711300e+10  
25%                    9.374400e+10  
50%                    9.777400e+10  
75%                    1.022070e+11  
max                    1.107630e+11

# Visualize

## Kết quả kinh doanh

In [10]:
vcb_kqkd["Tháng"] = vcb_kqkd["Kỳ"] * 3
vcb_kqkd

Năm  Kỳ  Doanh thu (đồng)  Tăng trưởng doanh thu (%)  \
0   2024   4    23581262000000                  -0.060425   
1   2024   3    23191212000000                  -0.125269   
2   2024   2    22862707000000                  -0.206768   
3   2024   1    24019660000000                  -0.132547   
4   2023   4    25097791000000                  -0.017087   
5   2023   3    26512406000000                   0.174704   
6   2023   2    28822211000000                   0.394328   
7   2023   1    27689870000000                   0.431878   
8   2022   4    25534087000000                   0.423362   
9   2022   3    22569428000000                   0.279451   
10  2022   2    20671039000000                   0.158584   
11  2022   1    19338145000000                   0.124238   
12  2021   4    17939277000000                   0.045207   

    Lợi nhuận sau thuế của Cổ đông công ty mẹ (đồng)  \
0                                      8565378000000   
1                                      8566598000000   
2                                      8119414000000   
3                                      8579996000000   
4                                      9355026000000   
5                                      7268919000000   
6                                      7422823000000   
7                                      8986435000000   
8                                      9928277000000   
9                                      6064870000000   
10                                     5936954000000   
11                                     7962184000000   
12                                     6539788000000   

    Tăng trưởng lợi nhuận (%)   LN trước thuế  Lợi nhuận thuần  \
0                   -0.084409  10702677000000    8569921000000   
1                    0.178524  10698944000000    8572494000000   
2                    0.093844  10116275000000    8124709000000   
3                   -0.045228  10718239000000    8585993000000   
4                   -0.057739  11693497000000    9360033000000   
5                    0.198528   9051164000000    7274556000000   
6                    0.250275   9277682000000    7427958000000   
7                    0.128639  11221344000000    8991901000000   
8                    0.518134  12419095000000    9933519000000   
9                    0.320135   7566222000000    6069444000000   
10                   0.499191   7423087000000    5941931000000   
11                   0.153509   9950256000000    7967426000000   
12                   0.153907   8174561000000    6545349000000   

    Cố tức đã nhận  Tháng  
0      66378000000     12  
1      49396000000      9  
2     159132000000      6  
3      32273000000      3  
4      50548000000     12  
5     146603000000      9  
6      29775000000      6  
7      39530000000      3  
8      32313000000     12  
9     127264000000      9  
10     24345000000      6  
11     24495000000      3  
12     31107000000     12

In [12]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["Doanh thu (đồng)"],
    mode="lines+markers",
    name="Doanh thu (đồng)"
))

fig.update_layout(
    title="Doanh thu (đồng) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Doanh thu (đồng)"
)

fig.show()

Theo biểu đồ, ta thấy:
  - Từ đầu năm 2022 đến quý 2 năm 2023, doanh thu có xu hướng tăng và đạt đỉnh ở quý 2 năm 2023.
  - Sau quý 2 năm 2023, doanh thu giảm dần, sau đó phục hồi dần từ quý 2 năm 2024.

In [14]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["Tăng trưởng doanh thu (%)"],
    marker_color=["green" if val > 0 else "red" for val in vcb_kqkd["Tăng trưởng doanh thu (%)"]]
))

fig.update_layout(
    title="Tăng trưởng doanh thu (%) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Tăng trưởng doanh thu (%)"
)

fig.show()


Theo biểu đồ tăng trưởng doanh thu, ta thấy:
  - Từ đầu năm 2022 - quý 3 năm 2023, tăng trưởng doanh thu dương. Trong đó, quý 1 năm 2023 tăng trưởng doanh thu đạt đỉnh.
  - Trong năm 2022, tăng trưởng doanh thu có xu hướng tăng.
  - Từ năm 2023, tăng trưởng doanh thu có xu hướng giảm. Trong đó từ cuối năm 2023 tăng trưởng doanh thu bắt đầu âm và đạt đáy vào quý 2 năm 2024.

In [15]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["Tăng trưởng lợi nhuận (%)"],
    marker_color=["green" if val > 0 else "red" for val in vcb_kqkd["Tăng trưởng lợi nhuận (%)"]]
))

fig.update_layout(
    title="Tăng trưởng lợi nhuận (%) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Tăng trưởng lợi nhuận (%)"
)

fig.show()


Theo biểu đồ tăng trưởng lợi nhuận, ta thấy:
- Từ năm 2022 đến quý 3 năm 2023, tăng trưởng lợi nhuận dương, trong đó quý 4 năm 2022 tăng trưởng đạt đỉnh.
- Tăng trưởng lợi nhuận từ năm 2023 có xu hướng giảm, trong đó từ cuối năm 2023 đến đầu năm 2024 tăng trưởng âm.
- Tăng trưởng lợi nhuận đạt đáy vào cuối năm 2024.

In [16]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["LN trước thuế"],
    mode="lines+markers",
    name="Lợi nhuận trước thuế (đồng)"
))
fig.add_trace(go.Scatter(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["Lợi nhuận thuần"],
    mode="lines+markers",
    name="Lợi nhuận thuần (đồng)"
))

fig.update_layout(
    title="Lợi nhuận trước thuế và lợi nhuận thuần theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Giá trị (đồng)"

)
fig.show()


Theo biểu đồ, ta thấy:
  - Quý 1/2022 - Quý 1/2023: LN trước thuế có sự biến động lớn. Ban đầu ở mức khoảng 100 nghìn tỷ đồng, sau đó giảm xuống mức thấp nhất khoảng 74 nghìn tỷ đồng vào giữa năm 2022, rồi tăng vọt lên mức cao nhất khoảng 124 nghìn tỷ đồng vào đầu năm 2023.
  - Quý 2/2023 - Quý 1/2025: Sau khi đạt đỉnh, LN trước thuế giảm xuống còn khoảng 92 nghìn tỷ đồng, sau đó tăng lên rồi lại giảm xuống, và cuối cùng có xu hướng tăng trở lại, đạt khoảng 107 nghìn tỷ đồng vào cuối kỳ.

In [17]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x=vcb_kqkd["Năm"].astype(str) + "-" + vcb_kqkd["Tháng"].astype(str),
    y=vcb_kqkd["Cố tức đã nhận"],
    name="Cổ tức đã nhận"
))

fig.update_layout(
    title="Cổ tức đã nhận theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Cổ tức đã nhận",
    xaxis_tickangle=-45
)

fig.show()


Theo biểu đồ, ta thấy cổ tức đã nhận có tính chu kì, luôn tăng cao vào quý 3 hàng năm.

## Cân đối kế toán

In [18]:
vcb_cdtc["Tháng"] = vcb_cdtc["Kỳ"] * 3
vcb_cdtc

Năm  Kỳ  TỔNG CỘNG TÀI SẢN (đồng)  TỔNG CỘNG NGUỒN VỐN (đồng)  \
0   2024   4          2085873522000000            2085873522000000   
1   2024   3          1932361908000000            1932361908000000   
2   2024   2          1905638840000000            1905638840000000   
3   2024   1          1773713584000000            1773713584000000   
4   2023   4          1839223339000000            1839223339000000   
5   2023   3          1731305572000000            1731305572000000   
6   2023   2          1704273133000000            1704273133000000   
7   2023   1          1846431290000000            1846431290000000   
8   2022   4          1814188209000000            1814188209000000   
9   2022   3          1649664327000000            1649664327000000   
10  2022   2          1601680964000000            1601680964000000   
11  2022   1          1462748158000000            1462748158000000   
12  2021   4          1414765480000000            1414765480000000   

    NỢ PHẢI TRẢ (đồng)  VỐN CHỦ SỞ HỮU (đồng)  LỢI ÍCH CỦA CỔ ĐÔNG THIỂU SỐ  \
0     1889664354000000        196209168000000                   96261000000   
1     1742064907000000        190297001000000                  110763000000   
2     1723936055000000        181702785000000                  105738000000   
3     1600099458000000        173614126000000                  100556000000   
4     1670680015000000        168543324000000                   93990000000   
5     1572062054000000        159243518000000                  104660000000   
6     1552272005000000        152001128000000                   98316000000   
7     1701772932000000        144658358000000                   93744000000   
8     1676199767000000        137988442000000                   88379000000   
9     1521274777000000        128389550000000                  102207000000   
10    1478691267000000        122989697000000                   97774000000   
11    1345670470000000        117077688000000                   92420000000   
12    1303594825000000        111170655000000                   87113000000   

    Tháng  
0      12  
1       9  
2       6  
3       3  
4      12  
5       9  
6       6  
7       3  
8      12  
9       9  
10      6  
11      3  
12     12

In [19]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_cdtc["Năm"].astype(str) + "-" + vcb_cdtc["Tháng"].astype(str),
    y=vcb_cdtc["TỔNG CỘNG TÀI SẢN (đồng)"],
    mode="lines+markers",
    name="Tổng cộng tài sản (đồng)"
))
fig.add_trace(go.Scatter(
    x=vcb_cdtc["Năm"].astype(str) + "-" + vcb_cdtc["Tháng"].astype(str),
    y=vcb_cdtc["NỢ PHẢI TRẢ (đồng)"],
    mode="lines+markers",
    name="Nợ phải trả (đồng)"
))

fig.update_layout(
    title="Tổng cộng tài sản và nợ phải trả theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Giá trị (đồng)"
)

fig.show()


Theo biểu đồ, ta thấy:
  - Tổng cộng tài sản và nợ phải trả có xu hướng tăng dần.
  - Vào quý 2 năm 2023 và quý 1 năm 2024 có xu hướng giảm nhưng sau đó hồi phục trở lại.

In [22]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_cdtc["Năm"].astype(str) + "-" + vcb_cdtc["Tháng"].astype(str),
    y=vcb_cdtc["VỐN CHỦ SỞ HỮU (đồng)"],
    mode="lines+markers",
    name="Vốn chủ sở hữu (đồng)"
))

fig.update_layout(
    title="Vốn chủ sở hữu theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Vốn chủ sở hữu (đồng)"
)

fig.show()


Theo biểu đồ, ta thấy vốn chủ sở hữu tăng dần theo thời gian.

In [23]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_cdtc["Năm"].astype(str) + "-" + vcb_cdtc["Tháng"].astype(str),
    y=vcb_cdtc["LỢI ÍCH CỦA CỔ ĐÔNG THIỂU SỐ"],
    mode="lines+markers",
    name="Lợi ích của cổ đông thiểu số"
))

fig.update_layout(
    title="Lợi ích của cổ đông thiểu số theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Lợi ích của cổ đông thiểu số"
)

fig.show()


Theo biểu đồ, ta thấy lợi ích cổ đông thiểu số có tính chu kì và tăng dần theo thời gian.

## Chỉ số tài chính

In [24]:
vcb_cstc["Tháng"] = vcb_cstc["Kỳ"] * 3
vcb_cstc

CP   Năm  Kỳ  Biên lợi nhuận ròng (%)   ROE (%)  ROIC (%)   ROA (%)  \
0   VCB  2024   4                 0.618781  0.187316       0.0  0.017237   
1   VCB  2024   3                 0.630934  0.193913       0.0  0.018586   
2   VCB  2024   2                 0.583807  0.195129       0.0  0.018386   
3   VCB  2024   1                 0.609459  0.199735       0.0  0.018516   
4   VCB  2023   4                 0.730792  0.219739       0.0  0.018083   
5   VCB  2023   3                 0.577076  0.226347       0.0  0.018943   
6   VCB  2023   2                 0.529423  0.230197       0.0  0.018477   
7   VCB  2023   1                 0.632716  0.231573       0.0  0.017892   
8   VCB  2022   4                 0.670401  0.244309       0.0  0.018522   
9   VCB  2022   3                 0.443859  0.220298       0.0  0.017240   
10  VCB  2022   2                 0.463926  0.216549       0.0  0.017014   
11  VCB  2022   1                 0.664853  0.207517       0.0  0.016503   
12  VCB  2021   4                 0.613101  0.215723       0.0  0.015994   

    Tỷ suất cổ tức (%)  Vốn hóa (Tỷ đồng)  Số CP lưu hành (Triệu CP)  \
0                  0.0    501340505640000                 8355675094   
1                  0.0    512519668725400                 5589091262   
2                  0.0    510842941346800                 5589091262   
3                  0.0    489604394551200                 5589091262   
4                  0.0    528728033385200                 5589091262   
5                  0.0    495752394939400                 5589091262   
6                  0.0    480661848532000                 5589091262   
7                  0.0    492181723384000                 4732516571   
8                  0.0    418354464876400                 4732516571   
9                  0.0    406049921791800                 4732516571   
10                 0.0    314712351971500                 4732516571   
11                 0.0    339794689797800                 4732516571   
12                 0.0    382860590593900                 4732516571   

          P/E       P/B       P/S  P/Cash Flow    EPS (VND)    BVPS (VND)  \
0   14.818799  2.556387  7.310462     8.479643  1025.097063  23470.623833   
1   14.803708  2.694830  7.669374   -10.421608  1532.735394  34028.114605   
2   15.329877  2.813058  7.767334   123.346014  1452.725250  32491.336872   
3   15.006220  2.821707  7.363946    -3.257646  1535.132564  31045.041468   
4   16.005957  3.205992  7.807162   -13.718398  1673.800903  29507.243355   
5   14.751702  3.115219  7.039211     6.687242  1300.554716  28473.118534   
6   14.834141  3.164272  6.735315    10.304262  1328.091214  27178.445454   
7   15.919692  3.404580  7.044594     1.981862  1898.870266  30547.090925   
8   13.992249  3.086173  6.144787     2.280837  2097.885311  28643.893786   
9   15.371801  3.165160  6.315958     8.228452  1281.531699  27107.637359   
10  12.616499  2.560887  5.173818     4.265981  1254.502528  25967.563168   
11  14.794486  2.904594  5.882990   -22.191178  1682.441864  24719.462942   
12  17.467214  3.511507  6.749586    -6.989035  1381.883804  23038.541622   

    Tháng  
0      12  
1       9  
2       6  
3       3  
4      12  
5       9  
6       6  
7       3  
8      12  
9       9  
10      6  
11      3  
12     12

In [26]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["ROE (%)"],
    mode="lines+markers",
    name="ROE (%)",
    yaxis="y1"
))
fig.add_trace(go.Scatter(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["ROA (%)"],
    mode="lines+markers",
    name="ROA (%)",
    yaxis="y2"))

fig.update_layout(
    title="ROE và ROA theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis=dict(
        title="ROE (%)",
        tickfont=dict(color="#1f77b4")
    ),
    yaxis2=dict(
        title="ROA (%)",
        overlaying="y",
        side="right",
        tickfont=dict(color="#ff7f0e")
    )
)

fig.show()


Đối với ROE:
- Từ 2022: ROE có xu hướng tăng trưởng đáng kể trong suốt năm 2022, đạt đỉnh vào khoảng cuối năm 2022.
- Năm 2023 - 2024: Sau khi đạt đỉnh, ROE lại có xu hướng giảm mạnh.

Đối với ROA:
- Đầu năm 2022: ROA cũng cho thấy sự tăng trưởng nhẹ.
- Từ cuối năm 2022 - 2024: ROA có xu hướng chu kỳ, sau đó giảm mạnh vào cuối năm 2024.

Công ty  ban đầu dùng đòn bẩy tài chính hợp lý, nhưng về sau có dấu hiệu mất kiểm soát tài chính hoặc lợi nhuận sụt giảm trong khi nợ vẫn cao (do ROE tụt xuống dưới ROA).

In [28]:
fig = go.Figure(data=[go.Bar(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["Biên lợi nhuận ròng (%)"],
    name="Biên lợi nhuận ròng (%)"
)])

fig.update_layout(
    title="Biên lợi nhuận ròng (%) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Biên lợi nhuận ròng (%)",
    xaxis_tickangle=-45
)

fig.show()


Theo biểu đồ, ta thấy:
- Quý 4 năm 2023: Biên lợi nhuận ròng đạt mức cao nhất.
- Quý 3 năm 2022: Mức biên lợi nhuận ròng thấp nhất.
- Không có xu hướng rõ ràng: Nhìn chung, không có một xu hướng tăng hoặc giảm rõ rệt nào của biên lợi nhuận ròng trong suốt giai đoạn này. Thay vào đó, nó dao động lên xuống đáng kể.

In [31]:
fig = go.Figure()

fig.add_trace(go.Scatter(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["Vốn hóa (Tỷ đồng)"],
    mode="lines+markers",
    name="Vốn hóa (tỷ đồng)"
))

fig.update_layout(
    title="Vốn hóa (tỷ đồng) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Vốn hóa (tỷ đồng)"
)

fig.show()


Theo biểu đồ, ta thấy:
- Vốn hóa có xu hướng tăng dần từ năm 2022 - 2024.
- Quý 2 năm 2022, quý 2 năm 2023 và quý 1 năm 2024: vốn hóa có giảm nhẹ nhưng sau đó phục hồi trở lại.

In [30]:
fig = go.Figure()

fig.add_trace(go.Bar(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["Số CP lưu hành (Triệu CP)"],
    name="Số cổ phiếu lưu hành (triệu cổ phiếu)"
))

fig.update_layout(
    title="Số cổ phiếu lưu hành (triệu cổ phiếu) theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="Số cổ phiếu lưu hành (triệu cổ phiếu)",
    xaxis_tickangle=-45
)

fig.show()


Từ biểu đồ, số cổ phiếu lưu hành có 2 lần điều chỉnh vào quý 2 năm 2023 và quý 4 năm 2024.

In [33]:
fig = make_subplots(specs=[[{"secondary_y": True}]])

fig.add_trace(
    go.Bar(
        x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
        y=vcb_cstc["BVPS (VND)"],
        name="BVPS"
    ),
    secondary_y=False
)
fig.add_trace(
    go.Scatter(
        x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
        y=vcb_cstc["EPS (VND)"],
        mode="lines+markers",
        name="EPS"
    ),
    secondary_y=True
)

fig.update_layout(
    title="BVPS và EPS theo thời gian",
    xaxis_title="Năm - Kỳ",
    yaxis_title="BVPS",
    yaxis2=dict(title="EPS", overlaying="y", side="right")
)

fig.show()


Về BVPS:
- BVPS có xu hướng tăng nhẹ theo chu kỳ.
- Mức tăng/giảm không quá lớn nhưng cho thấy giá trị tài sản ròng trên mỗi cổ phiếu có xu hướng đi lên.

Về EPS:
- EPS có sự biến động đáng kể hơn so với BVPS.
- Từ năm 2022, EPS có sự biến động mạnh. EPS đạt đỉnh vào khoảng cuối năm 2022.
- Từ năm 2023 trở đi, EPS có xu hướng giảm dần và đạt đáy vào cuối năm 2024.

In [ ]:
fig = make_subplots(
    rows=3,
    cols=1,
    shared_xaxes=True,
    vertical_spacing=0.03,
    subplot_titles=("P/E theo thời gian", "P/B theo thời gian", "P/S theo thời gian")
)

fig.add_trace(
    go.Scatter(
        x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
        y=vcb_cstc["P/E"], name="P/E"
    ),
    row=1,
    col=1
)
fig.add_trace(
    go.Scatter(
        x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
        y=vcb_cstc["P/B"], name="P/B"
    ),
    row=2,
    col=1
)
fig.add_trace(
    go.Scatter(
        x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
        y=vcb_cstc["P/S"], name="P/S"
    ),
    row=3,
    col=1
)

fig.update_layout(height=800, width=800, title_text="P/E, P/B, và P/S Ratios")
fig.update_xaxes(tickangle=-45, tickfont=dict(size=10), row=3, col=1)

fig.show()

Nhận xét về biểu đồ P/E:
- Biến động: Tỷ số P/E có sự biến động khá lớn trong giai đoạn này.
- Đỉnh và đáy: Có thể thấy nhiều đỉnh và đáy, cho thấy sự thay đổi trong kỳ vọng của thị trường về lợi nhuận của công ty.
- Xu hướng chung: Không có một xu hướng tăng hoặc giảm rõ rệt trong toàn bộ giai đoạn, mà là sự dao động lên xuống. Tuy nhiên, có vẻ như P/E có xu hướng tăng nhẹ vào cuối giai đoạn được thể hiện.

Nhận xét về biểu đồ P/B:
- Biến động: Tỷ số P/B cũng cho thấy sự biến động, nhưng có vẻ ít hơn so với P/E.
- Tăng trưởng và giảm: P/B có xu hướng tăng từ giữa năm 2022 và đạt đỉnh đầu năm 2023, sau đó giảm mạnh và đạt đáy vào cuối năm 2024.

Nhận xét về biểu đồ P/S:
- Xu hướng tăng rõ rệt: Tỷ số P/S cho thấy một xu hướng tăng khá rõ rệt từ giữa năm 2022 đến năm 2024.
- Ổn định vào năm 2024: Tỷ số P/S loanh quanh ở giá trị 7.5

Trong khi P/S có xu hướng tăng, P/E và P/B lại cho thấy sự biến động với các đỉnh và đáy khác nhau. Điều này có thể phản ánh sự thay đổi trong kỳ vọng của thị trường về các khía cạnh khác nhau của công ty (lợi nhuận, giá trị tài sản, doanh thu).

In [ ]:
fig = go.Figure(data=[go.Bar(
    x=vcb_cstc["Năm"].astype(str) + "-" + vcb_cstc["Tháng"].astype(str),
    y=vcb_cstc["P/Cash Flow"],
    marker_color=["red" if value < 0 else "green" for value in vcb_cstc["P/Cash Flow"]]
)])

fig.update_layout(
    title="P/Cash Flow theo thời gian",
    xaxis_title="Năm - Tháng",
    yaxis_title="P/Cash Flow",
    xaxis_tickangle=-45
)

fig.show()


Dựa trên biểu đồ, ta thấy:
- Đỉnh điểm bất thường: Vào tháng 6 năm 2024, tỷ số P/Cash Flow tăng vọt lên mức 123.35. Đây là một giá trị rất cao, cho thấy nhà đầu tư sẵn sàng trả một mức giá rất cao cho mỗi đơn vị dòng tiền của công ty vào thời điểm đó. Cần xem xét kỹ lưỡng các yếu tố nào đã dẫn đến sự tăng đột biến này.
- Xu hướng không rõ ràng: Không có một xu hướng tăng hoặc giảm rõ rệt nào của tỷ số P/Cash Flow trong suốt giai đoạn này, mà là sự biến động thất thường.